Cql/CodeGeneration.NET/ExpressionConverter.cs

```diff
@@ -20,13 +20,8 @@
 
 namespace Hl7.Cql.CodeGeneration.NET
 {
-    internal class ExpressionConverter
+    internal class ExpressionConverter(string libraryName)
```

Comment:
I'm not convinced we should use primary constructors for non-poco types, because we have different naming convention between parameters and fields, and this forces them to be the same

Reply:
Here 'libraryName' is both the parameter name in the constructor and also the field in the class. Fields start with and underscore, so this breaks convention. Furthermore, primary class constructors do not generate `readonly` fields, so any member in the class can modify it and it should not be able to do that.

Reply:
I see them as parameters to the class that can be accessed everywhere - so they are kind of "new" - they are neither fields nor properties.  Agree that it would be better if they were readonly, but I don't think you can say that of any parameter to any function in C#.

Reply:
Solution to making `libraryName` readonly, is to add a field explicitly.

![image](https://github.com/FirelyTeam/firely-cql-sdk/assets/4639799/97996491-4709-4657-8a20-6af2b4b9365d)

Reply:
You wouldn't do that with functions either.... and Rider will do its utmost to refactor that away again. I think it's not worth the cost of adding boilerplate.  We got used to it for functions too after all.

Reply:
Ok, I'm hoping C# vNext solves this somehow, allowing modifier keywords to be used in primary constructor parameters such as `readonly`. I'll resolve this for now.

Reply:
It's being discussed: https://github.com/dotnet/csharplang/discussions/7699